pkg/apiserver/user/sqlauth/sqlauth.go

```diff
@@ -50,6 +65,29 @@ func (a *Authenticator) Authenticate(f user.AuthenticateForm) (*utils.SessionUse
 	}
 	defer utils.CloseTiDBConnection(db) //nolint:errcheck
 
+	// Check dashboard privileges
+	// 1. Check whether TiDB SEM is enabled
+	resData, err := a.tidbClient.SendGetRequest("/config")
+	if err != nil {
+		return nil, user.ErrSignInOther.WrapWithNoMessage(err)
+	}
+	var config tidbSecurityConfig
+	err = json.Unmarshal(resData, &config)
```

Comment:
Will this cause issues when running against a TiDB that does not have SEM feature (like 4.0.x)?

Reply:
I have tested it by the following code:

```
	var config tidbSecurityConfig
	_ = json.Unmarshal([]byte(""), &config)
	fmt.Println("sem:", config.Security.EnableSem)
```

It will output:

```
sem: false
```